internal/core/metadata/rest_provisionwatcher.go

```diff
@@ -478,16 +478,21 @@ func restDeleteProvisionWatcherById(
 	// Check if the provision watcher exists
 	pw, err := dbClient.GetProvisionWatcherById(id)
 	if err != nil {
-		errorHandler.Handle(w, err, errorconcept.ProvisionWatcher.NotFoundById)
+		errorHandler.HandleOneVariant(
+			w,
+			err,
+			errorconcept.ProvisionWatcher.NotFoundById,
+			errorconcept.Default.InternalServerError)
 		return
 	}
 
-	err = deleteProvisionWatcher(pw, w, lc, dbClient, errorHandler)
-	if err != nil {
-		errorHandler.Handle(w, err, errorconcept.ProvisionWatcher.DeleteError_StatusInternalServer)
+	if err = deleteProvisionWatcher(pw, w, lc, dbClient, errorHandler); err != nil {
+		lc.Error("Problem deleting provision watcher: " + err.Error())
 		return
```

Comment:
Is this change intentional? It looks like this may change the existing behavior. Before this error would result in a `404 Not Found` and with these changes we are logging and returning which would result in a `200 OK` by default.